solutions/platinum/cf-600E.mdx

```diff
@@ -0,0 +1,80 @@
+---
+id: cf-600E
+source: CF
+title: Lomsat gelral
+author: Chongtian Ma
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/21827)
+
+## Implementation
+
+As explained in the official analysis, when we use small to large merging on data structures such as sets and maps, we optimize the amount of operations from $$\mathcal{O}(N^2 \log N)$ to $\mathcal{O}(N \log^2 N)$. We can take advantage of this to store all kinds of information related to colors, their sum, and their occurrences.
+
+**Time Complexity:** $\mathcal{O}(N \log^2 N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+const int MAXN = 1e5 + 1;
+array<ll, MAXN> color, ans;
+vector<int> g[MAXN];
+
+// col_cnt[i][j] = number of occurrences of color j in i
+map<ll, ll> col_cnt[MAXN];
```

Comment:
rename `col_occs` or smth

Reply:
same thing???????

Reply:
then expand it to `col_count`

Reply:
same thing??
what else could cnt stand for?

Reply:
we had this conversation before

Reply:
and..?

Reply:
you agreed to rename it

Reply:
can you inform me why you want to rename it again?

Reply:
are you guys really arguing about a name change LOL